notation.go

```diff
@@ -52,6 +52,17 @@ type Signer interface {
 // remote.
 // The descriptor of the sign content is returned upon sucessful signing.
 func Sign(ctx context.Context, signer Signer, repo registry.Repository, opts SignOptions) (ocispec.Descriptor, error) {
+	// Input validation for expiry duration
+	if opts.ExpiryDuration != nil {
+		expDurInSec := opts.ExpiryDuration.Seconds()
+		if expDurInSec != float64(int64(expDurInSec)) {
```

Comment:
Converting from `int64` to `float64` then to `int64` and then to `float64` seems not right.

Suggesting
```suggestion
		if opts.ExpiryDuration % time.Second != 0 {
```